src/utils/i18n.tsx

```diff
@@ -58,8 +53,8 @@ export class I18n {
     * Check if the given language is supported.
     * @param language The language to check, for example 'en'.
     */
-    public static isLanguageSupported(language: Languages): boolean {
-        return I18n._supportedLanguages.indexOf(language) > -1;
+    public static isLanguageSupported(language: string): boolean {
+        return I18n._supportedLanguages.indexOf(language) > -1 && language !== undefined;
```

Comment:
I18n._supportedLanguages.indexOf(undefined) == -1 right, why do we need `language !== undefined`?
Note comparing to undefined doesn't work always, you need to use [typeof language == 'undefined'](https://stackoverflow.com/questions/2778901/how-to-compare-variables-to-undefined-if-i-don-t-know-whether-they-exist)

Reply:
You're right, that slipped through when i was testing.

Reply:
Comparing to `undefined` is fine here. TypeScript should prevent you from trying to access a variable which hasn't been defined, which is what Stack Overflow was talking about.

Reply:
`undefined` is not a keyword in JS, if any library does `undefined=1` this suddenly breaks. It shouldn't happen though..

Reply:
The global `undefined` variable has been read only since the Firefox 4 and ES5 era. You can't overwrite it.

Reply:
Didn't know that, thanks!